include/kodlab_mjbots_sdk/behavior.h

```diff
@@ -0,0 +1,205 @@
+   *          a good place to read in the initial robot state and set the
+   *          appropriate variables to begin the behavior. The previously active
+   *          behavior is passed as an argument, permitting different
+   *          functionalities for different prior behaviors.
+   * @param prev_behavior previously active behavior
+   */
+  virtual void Begin(const Behavior<Robot> &prev_behavior) {}
+
+  /**
+   * @brief Update this behavior for a given robot interface
+   * @note This function must be implemented in a derived class
+   * @details This code runs after Begin, and is called on every control loop
+   *          update. Use this to define the primary content of your behavior,
+   *          e.g., to update and command new joint torques.  Additionally, this
+   *          code can monitor if the behavior has terminated and update flags
+   *          appropriately.
+   * @param robot robot interface
+   */
+  virtual void Update() = 0;
+
+  /**
+   * @brief Stop this behavior
+   * @details This code is run when a behavior is directed to stop running by
+   *          the behavior manager. This should trigger a shutdown process for
+   *          the behavior, e.g. slowing the robot motion. The upcoming behavior
+   *          is passed as an argument to permit different shutdown procedures
+   *          for different on-deck behaviors.
+   * @param next_behavior behavior to be transitioned to
+   */
+  virtual void Stop(const Behavior<Robot> &next_behavior) {}
+
+  /**
+   * @brief Check for whether the behavior is running
+   * @details This function identifies if the behavior is active. By default, it
+   *          simply accesses the `active_` variable which is toggled by the
+   *          behavior manager.
+   * @return true if this behavior is active, false otherwise
+   */
+  virtual bool Running() { return active_; }
+
+  /**
+   * @brief Check for whether this behavior is prepared to switch to a new
+   *        behavior
+   * @details This function indicates to the behavior manager whether the
+   *          behavior is prepared to terminate and transition to the next
+   *          behavior. In its simplest form, this function accesses the
+   *          internal behavior active flag. This function can be used as a mode
+   *          switch guard.
+   * @param next_behavior behavior being transitioned to
+   * @return true if behavior is ready to switch, false otherwise
+   */
+  virtual bool ReadyToSwitch(const Behavior<Robot> &next_behavior)
+  {
+    return true;
+  }
+
+  /**
+   * @brief Processes input data, if available
+   * @details This method should be implemented by child classes wishing to
+   * handle input processing internally. For supported LCM behavior inputs,
+   * please inherit `IOBehavior`.
+   * @note If using a behavior manager, this method is called every update
+   * cycle, and should implement checking to determine if a new input is
```

Comment:
Who is responsible for checking if a new input is available prior to using the data?

Reply:
If this method is being overridden by the user in a derived class, then it is up to the user to handle checking for whatever their input method is.  In our case, using LCM inputs, this method is overridden in `IOBehavior` using the built-in `LcmMessageHandler` functionalities to check for inputs and process new input, if desired.